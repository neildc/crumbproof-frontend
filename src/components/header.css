.navLinks {
    display: flex;
    align-items: center;
}

.drawer > div {
    top: 50px !important;
}

@media (min-width:640px) {

    .appBar {
        padding: 0px 10% !important;
    }

    .authButton {
        margin-left: 100px !important;
    }
}


.tooltip {
    font-size: 14px !important;
    display: block !important;
}
