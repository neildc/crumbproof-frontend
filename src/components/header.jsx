import _ from 'lodash';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Drawer from 'material-ui/Drawer';
import { ListItem } from 'material-ui/List';

import AppBar from 'material-ui/AppBar';
import FlatButton from 'material-ui/FlatButton';

import RecipeIcon from 'material-ui/svg-icons/action/description';
import ActivityIcon from 'material-ui/svg-icons/image/gradient';
import LiveIcon from 'material-ui/svg-icons/image/timer';

import { authLogout } from '../actions/actions_auth';

import './header.css';


const navItems = [
  { title: 'Activities', url: '/activity', icon: ActivityIcon },
  { title: 'Recipes', url: '/recipes', icon: RecipeIcon },
  { title: 'Live activity', url: '/live', icon: LiveIcon },
];

export class Header extends Component {
  constructor(props) {
    super(props);
    this.state = { drawerOpen: false };
  }

  handleDrawerToggle() {
    this.setState({ drawerOpen: !this.state.drawerOpen });
  }

  authButton() {
    const Button = props => (
      <FlatButton
        {...props}
        labelStyle={{ color: 'white' }}
        className="authButton"
      />
    );

    if (this.props.authenticatedUser) {
      return <Button onClick={this.props.authLogout} label="Logout" />;
    }
    return <Button containerElement={<Link to="/login" />} label="Login" />;
  }

  closeDrawer() {
    this.setState({ drawerOpen: false });
  }

  renderAppBarNavLinks() {
    return (
      _.map(navItems, i => (
        <FlatButton
          key={i.title}
          containerElement={<Link to={i.url} />}
          label={i.title}
          labelStyle={{ color: 'white' }}
          icon={<i.icon color="#FFFFFF" />}
        />
      ))
    );
  }

  renderNavDrawerItems() {
    return (
      _.map(navItems, i => (
        <ListItem
          key={i.title}
          primaryText={i.title}
          containerElement={<Link to={i.url} />}
          leftIcon={<i.icon />}
          onClick={() => this.closeDrawer()}
        />
      ))
    );
  }

  render() {
    return (
      <div style={{ position: 'fixed', width: '100%', zIndex: 3 }}>
        <AppBar
          className="appBar"
          title="crumbproof"
          onLeftIconButtonTouchTap={() => this.handleDrawerToggle()}
        >
          <div className="navLinks">
            { window.innerWidth > 640 &&
              this.renderAppBarNavLinks()
            }
            {this.authButton()}
          </div>
        </AppBar>

        <Drawer
          open={this.state.drawerOpen}
          docked={false}
          className="drawer"
          onRequestChange={open => this.setState({ drawerOpen: open })}
        >
          {this.renderNavDrawerItems()}
        </Drawer>
      </div>
    );
  }
}

function mapStateToProps({ auth }) {
  return {
    authenticatedUser: auth.user,
  };
}

export default connect(mapStateToProps, { authLogout })(Header);
