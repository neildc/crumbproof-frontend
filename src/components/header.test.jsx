import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header';

it('renders without throwing', () => {
  const component = shallow(
    <Header />,
  );

  expect(component).toMatchSnapshot();
});
