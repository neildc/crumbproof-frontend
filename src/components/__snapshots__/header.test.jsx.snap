// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without throwing 1`] = `
<div
  style={
    Object {
      "position": "fixed",
      "width": "100%",
      "zIndex": 3,
    }
  }
>
  <AppBar
    className="appBar"
    onLeftIconButtonTouchTap={[Function]}
    showMenuIconButton={true}
    title="crumbproof"
    zDepth={1}
  >
    <div
      className="navLinks"
    >
      <FlatButton
        containerElement={
          <Link
            replace={false}
            to="/activity"
          />
        }
        disabled={false}
        fullWidth={false}
        icon={
          <ImageGradient
            color="#FFFFFF"
          />
        }
        key="Activities"
        label="Activities"
        labelPosition="after"
        labelStyle={
          Object {
            "color": "white",
          }
        }
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
      <FlatButton
        containerElement={
          <Link
            replace={false}
            to="/recipes"
          />
        }
        disabled={false}
        fullWidth={false}
        icon={
          <ActionDescription
            color="#FFFFFF"
          />
        }
        key="Recipes"
        label="Recipes"
        labelPosition="after"
        labelStyle={
          Object {
            "color": "white",
          }
        }
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
      <FlatButton
        containerElement={
          <Link
            replace={false}
            to="/live"
          />
        }
        disabled={false}
        fullWidth={false}
        icon={
          <ImageTimer
            color="#FFFFFF"
          />
        }
        key="Live activity"
        label="Live activity"
        labelPosition="after"
        labelStyle={
          Object {
            "color": "white",
          }
        }
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        primary={false}
        secondary={false}
      />
      <Button
        containerElement={
          <Link
            replace={false}
            to="/login"
          />
        }
        label="Login"
      />
    </div>
  </AppBar>
  <Drawer
    className="drawer"
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={30}
    width={null}
    zDepth={2}
  >
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement={
        <Link
          replace={false}
          to="/activity"
        />
      }
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="Activities"
      leftIcon={<ImageGradient />}
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Activities"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement={
        <Link
          replace={false}
          to="/recipes"
        />
      }
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="Recipes"
      leftIcon={<ActionDescription />}
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Recipes"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
    <ListItem
      autoGenerateNestedIndicator={true}
      containerElement={
        <Link
          replace={false}
          to="/live"
        />
      }
      disableKeyboardFocus={false}
      disabled={false}
      initiallyOpen={false}
      insetChildren={false}
      key="Live activity"
      leftIcon={<ImageTimer />}
      nestedItems={Array []}
      nestedLevel={0}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onNestedListToggle={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      primaryText="Live activity"
      primaryTogglesNestedList={false}
      secondaryTextLines={1}
    />
  </Drawer>
</div>
`;
